utils/http_utils.go

```diff
@@ -22,52 +23,59 @@ type HttpResponse struct {
 	StatusCode int
 }
 
-// It send an http post to the endpoint signing the body with the worker's private key
-func SignedPost(workerId string, body interface{}, endpoint string) *HttpResponse{
-	requestBody, err := json.Marshal(body)
+func getSignature(payload interface{}, workerId string) []byte {
+	parsedPayload, err := json.Marshal(payload)
 
 	if err != nil {
-		log.Fatal("Error on marshalling the request body")
+		log.Fatal("Error on marshalling the payload")
 	}
 
-	data, hashSum := SignMessage(GetPrivateKey(workerId), requestBody)
+	signedPayload, _ := SignMessage(GetPrivateKey(workerId), parsedPayload)
 
-	payload := &map[string][]byte{"data": data, "hashSum": hashSum}
+	return signedPayload
+}
 
-	return Post(payload, endpoint)
+func AddSignature(workerId string, payload interface{}, headers http.Header) http.Header {
+	signature := GetSignature(payload, workerId)
+	headers.Set("SIGNATURE", string(signature))
+	return headers
 }
 
-func Post(body interface{}, endpoint string) *HttpResponse{
+func Post(workerId string, body interface{}, headers http.Header, endpoint string) (*HttpResponse, error){
+	headers = AddSignature(workerId, body, headers)
+
 	requestBody, err := json.Marshal(body)
 
 	if err != nil {
 		log.Fatal("Unable to marshal body")
 	}
 
 	req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewBuffer(requestBody))
-
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
 
+	req.Header = headers
```

Comment:
Perhaps replacing is not the best choice. I think that maybe we just want to overwrite headers or add new ones. What do you think? If you agree, I suggest adding a comment to resolve this issue later.

Reply:
but the variable headers is the composition of the received param + the changes made by AddSignature, idk if it address your point.